src/components/SigningModals/MultiScreenModal.js

```diff
@@ -163,6 +168,28 @@ function MultiScreenModal({ visible, screens, onClose }) {
   )
 }
 
+function useScreens(screens) {
+  const { direction, next, prev, step } = useSteps(screens.length)
+  const [screensState, setScreensState] = useState(screens)
+
+  useEffect(() => {
+    setScreensState(screens)
+  }, [screens])
+
+  const getScreen = useCallback(step => screensState[step], [screensState])
+
+  const currentScreen = getScreen(step)
```

Comment:
Small thing, but should we wrap this with a `useMemo()`?

Reply:
Can do, I wonder at what point this type of optimisation is helping though. My understanding is `useMemo()` needs to make an additional function call on top of our existing call to make an assignment.

Reply:
Yes, good point. In this case, I mainly wanted to ensure that `currentScreen` wasn't changing all the time as I suspected that it would be driving the modal.

However, the [actual implementation uses `getScreen()` for rendering the contents](https://github.com/aragon/aragon/blob/cbb03eec471132ec33122be28e0e7485b2dc34c9/src/components/SigningModals/MultiScreenModal.js#L135) due to the transition, so this optimization is definitely not very useful.